esp/bindings/http/platform/httpbinding.cpp

```diff
@@ -271,6 +270,92 @@ EspHttpBinding::EspHttpBinding(IPropertyTree* tree, const char *bindname, const
             memCachedInitString.set("--SERVER=127.0.0.1");//using local memcached server
     }
 #endif
+    if(!m_secmgr.get())
+        return;
+
+    processName.set(procname);
+    const char* authDomain = bnd_cfg->queryProp("@authDomain");
+    if (!isEmptyString(authDomain))
+        domainName.set(authDomain);
+    else
+        domainName.set("default");
+
+    VStringBuffer xPath("%s/%s[@name=\"%s\"]/%s[@name=\"%s\"]", PathSessionRoot, PathSessionProcess,
+            procname, PathSessionDomain, domainName.str());
+    VStringBuffer appStr("%s[@port=\"%d\"]", PathSessionApplication, m_port);
+    Owned<IRemoteConnection> conn = querySDS().connect(xPath.str(), myProcessSession(), RTM_LOCK_WRITE, SESSION_SDS_LOCK_TIMEOUT);
+    if (!conn)
+        throw MakeStringException(-1, "Failed to connect SDS DomainSession.");
+    IPropertyTree* sessionDomain = conn->queryRoot();
```

Comment:
I will create two new methods to include most of those code lines. Similar code in ws_espcontrolservice.cpp will be moved to a new method. But, I may not share the method here.